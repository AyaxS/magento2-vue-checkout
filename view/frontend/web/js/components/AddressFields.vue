<template>
  <div>
    <BaseInput
      v-model="address.firstname"
      label="First name"
      name="firstname"
      type="text"
      validate-type="required"
    />
    <BaseInput
      v-model="address.lastname"
      label="Last name"
      name="lastname"
      type="text"
      validate-type="required"
    />
    <BaseInput
      v-model="address.telephone"
      label="Phone Number"
      name="telephone"
      type="tel"
      validate-type="required"
    />
    <BaseInput
      v-model="address.street0"
      label="Street Address"
      name="street[0]"
      type="text"
      validate-type="required"
    />
    <BaseInput
      v-model="address.street1"
      label="Street Address"
      name="street[1]"
      type="text"
    />
    <div>
      <label for="country">
        Select Country
      </label>
      <multiselect
        id="country"
        v-model="address.country_id"
        :options="countries"
        :allow-empty="false"
        :show-labels="false"
        data-vv-as="Country"
        name="country"
        label="label"
        placeholder="Select country"
        @input="onCountryChange"
      />
    </div>
    <BaseInput
      v-model="address.city"
      label="City"
      name="city"
      type="text"
      validate-type="required"
    />
    <BaseInput
      v-model="address.postcode"
      label="Zip/Postal Code"
      name="postcode"
      type="text"
      validate-type="required"
    />
    <BaseInput
      v-if="!regions.length"
      v-model="address.region"
      :validate-type="!regions.length ? 'required' : ''"
      label="State/Province"
      name="region"
      type="text"
    />
    <div
      v-if="regions.length"
    >
      <label for="region_id">
        Select State/Province
      </label>
      <multiselect
        id="region_id"
        v-model="address.region_id"
        :options="regions"
        :allow-empty="false"
        :show-labels="false"
        data-vv-as="Region"
        name="region_id"
        label="label"
        placeholder="Select State/Province"
      />
    </div>
    <BaseInput
      v-model="address.company"
      label="Company"
      name="company"
      type="text"
    />
  </div>
</template>

<script>
import BaseInput from './BaseInput.vue'
import countries from '../data/countries.json'
import Multiselect from 'vue-multiselect'
import EventBus from '../event-bus'

export default {
  components: {
    BaseInput,
    Multiselect
  },
  props: {
    type: {
      type: String,
      required: true
    }
  },
  data () {
    return {
      countries
    }
  },
  computed: {
    addressData () {
      return this.$store.getters.addressByType(this.type)
    },
    regions () {
      if (this.address.country_id.value === null) {
        return []
      } else {
        return this.$store.getters.regionsByCountryId(this.address.country_id.value)
      }
    }
  },
  created () {
    // this.address = this.addressData
    console.log(this.address)
    EventBus.$once('save-address', (type) => {
      this.$store.commit('setAddress', { type, address: this.address })
    })
  },
  methods: {
    onCountryChange () {
      if (this.type === 'shippingAddress') {
        this.$store.dispatch('updateShippingMethods', this.address.country_id.value)
      }
    }
  }
}
</script>

<style lang="scss">
  .input {
    &--error {
      & .input__message {
        display: block;
        color: red;
      }
    }
  }

  input[type="text"] {
    &.multiselect__input {
      height: auto;
      border: none;
    }
  }

</style>
