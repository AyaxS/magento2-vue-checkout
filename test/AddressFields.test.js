import { shallowMount, createLocalVue } from '@vue/test-utils'
import AddressFields from '../view/frontend/web/js/components/AddressFields.vue'
import VeeValidate from 'vee-validate'

const localVue = createLocalVue()
localVue.use(VeeValidate)

describe('AddressFields.test.js', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(AddressFields, {
      propsData: {
        type: 'shippingAddress'
      },
      computed: {
        addressData: () => {
          return {
            city: 'Arizona',
            company: 'Cafe',
            country_id: {
              is_region_visible: true,
              label: 'United States',
              value: 'US',
            },
            firstname: 'John',
            lastname: 'Lennon',
            postcode: '12312',
            region: 'Arizona',
            region_id: {
              country_id: 'US',
              label: 'Arizona',
              title: 'Arizona',
              value: '4'
            },
            street0: 'Sample Street',
            street1: '2',
            telephone: '123123123',
          }
        },
        regions: () => {
          return [
            {
              country_id: 'US',
              label: 'Arizona',
              title: 'Arizona',
              value: '4'
            }
          ]
        }
      },
      localVue
    })
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
