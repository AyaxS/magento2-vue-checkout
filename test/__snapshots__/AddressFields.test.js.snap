// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressFields.test.js has the expected html structure 1`] = `
<div>
  <baseinput-stub />
   
  <baseinput-stub />
   
  <baseinput-stub />
   
  <baseinput-stub />
   
  <baseinput-stub />
   
  <div
    class=""
  >
    <label
      for="country"
    >
      
      Select Country
    
    </label>
     
    <vue-multiselect-stub
      aria-invalid="false"
      aria-required="true"
      data-vv-as="Country"
      id="country"
      label="label"
      options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      placeholder="Select country"
    />
     
    <span
      class="input__message"
      style="display: none;"
    >
      
      
    
    </span>
  </div>
   
  <baseinput-stub />
   
  <baseinput-stub />
   
  <!---->
   
  <div
    class=""
  >
    <label
      for="region_id"
    >
      
      Select State/Province
    
    </label>
     
    <vue-multiselect-stub
      aria-invalid="false"
      aria-required="true"
      data-vv-as="Region"
      id="region_id"
      label="label"
      options="[object Object]"
      placeholder="Select State/Province"
    />
     
    <span
      class="input__message"
      style="display: none;"
    >
      
      
    
    </span>
  </div>
   
  <baseinput-stub />
</div>
`;
